### PR TITLE
test: qbo advanced settings

### PR DESCRIPTION
**Pull request by anishfyle**
Screenshot 2024-09-20 at 6 02 57 PM

### Clickup Link
https://app.clickup.com/t/86cwc2w00

### CODE REVIEW

File: src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts [hunk 1]
```diff
@@ -1,23 +1,572 @@
...
+        CLASS: [],
+        ITEM: [],
+        PAYMENT_ACCOUNT: [],
+        EMPLOYEE: [],
+        JOB: [],
+        CREDIT_CARD_ACCOUNT: [],
+        ACCOUNTS_PAYABLE: [],
+        TAX_CODE: [],
+        COMPANY: []
+      }));
```
Comment (DhaaraniCIT) [inline src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts:115]: get this data from fixture, like others

File: src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts [hunk 2]
```diff
@@ -1,23 +1,572 @@
...
+      expect(component.advancedSettingForm).toBeDefined();
+      expect(component.skipExportForm).toBeDefined();
+      expect(component['getSettingsAndSetupForm']).toHaveBeenCalled();
+      expect(component['setupFormWatchers']).toHaveBeenCalled();
+      expect(component.isOnboarding).toBeTrue();
+    }));
+
+    it('should concatenate additional email options', fakeAsync(() => {
+      const mockAdditionalEmails = [
+        { name: 'Additional User', email: 'additional@example.com' }
```
Comment (DhaaraniCIT) [inline src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts:137]: move the const to fixture

File: src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts [hunk 3]
```diff
@@ -1,23 +1,572 @@
...
+      ];
+      const mockAdvancedSettingsWithAdditionalEmails = {
+        ...mockQboAdvancedSettings,
```
Comment (DhaaraniCIT) [inline src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts:140]: same as above

File: src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts [hunk 4]
```diff
@@ -1,23 +1,572 @@
...
+        value1: ['anish']
+      });
+      component.expenseFilters = { count: 0, next: null, previous: null, results: [] };
+    });
+  
+    it('should save advanced settings and skip export fields successfully', fakeAsync(() => {
+      advancedSettingsService.postAdvancedSettings.and.returnValue(of(mockQboAdvancedSettings));
+      skipExportService.postExpenseFilter.and.returnValue(of({
+        ...mockExpenseFilter,
+        id: 75,
```
Comment (DhaaraniCIT) [inline src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts:187]: move to fixture

File: src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts [hunk 5]
```diff
@@ -1,23 +1,572 @@
...
+        name_in_journal_entry: NameInJournalEntry.EMPLOYEE
+      } as any;
+      expect(component.isAutoCreateMerchantsAsVendorsFieldVisible()).toBeFalse();
+    });
+  });
+
+  describe('onMultiSelectChange', () => {
+    it('should update memoStructure with formatted memo', () => {
+      const mockMemo = ['employee_email', 'merchant', 'purpose'];
+      const mockFormattedMemo = ['employee_email', 'merchant', 'purpose', 'category'];
```
Comment (DhaaraniCIT) [inline src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts:385]: same as above

File: src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts [hunk 6]
```diff
@@ -1,23 +1,572 @@
...
+      component.advancedSettingForm = new FormBuilder().group({
+        memoStructure: [mockMemo]
+      });
+      component.defaultMemoOptions = ['employee_email', 'merchant', 'purpose', 'category', 'spent_on'];
```
Comment (DhaaraniCIT) [inline src/app/integrations/qbo/qbo-shared/qbo-advanced-settings/qbo-advanced-settings.component.spec.ts:389]: can we declare it in the beginning itself as we use it multiple times and it Is not modified

**Review by ashwin1111 — APPROVED** [review]

**Review by DhaaraniCIT — APPROVED** [review]

**Pull request merged**